syftlib/src/main/java/org/openmined/syft/datasource/RandomSampler.kt

```diff
@@ -0,0 +1,34 @@
+package org.openmined.syft.datasource
+
+import java.util.stream.IntStream.range
+import org.pytorch.Tensor
+
+/**
+ * Samples elements randomly. If without replacement, then sample from a shuffled dataset.
+ * If with replacement, then user can specify :attr:`num_samples` to draw.
+ * @param dataSource (Dataset): dataset to sample from
+ * @param replacement (Boolean): samples are drawn on-demand with replacement if ``True``, default=``False``
+ * @param numSamples (Int): number of samples to draw, default=`len(dataset)`. This argument
+ *                          is supposed to be specified only when `replacement` is ``True``.
+ */
+class RandomSampler(override var dataSource: Dataset, var replacement: Boolean = false,
+                    var numSamples: Int = dataSource.len()) : Sampler(dataSource) {
+
+     fun numsamples() : Int{
+
+         if (numSamples == null) {
```

Comment:
Use `numSample?.let {} :? `

Reply:
like `numSamples?.let {return dataSource.len()}`  ?

Reply:
> Use `numSample?.let {} :? `

like `numSamples?.let {return dataSource.len()}` ?

Reply:
It's more Kotlin idiomatic. Also the last condition should never happen (`return 0`).
```
fun getNumSamples() : Int{
	return numSamples?.let {
        numSamples
    } ?: 
        dataSource.size
}
```